### PR TITLE
Add package-refs to reference assemblies for linux build

### LINKED ISSUE #1326: BDN does not build using dotnet sdk from the command line in Linux
When checking out master copy of BDN in Linux and issuing `dotnet build` the build fails due to missing reference assemblies on Linux.

```bash
...
XUnit/TheoryWindowsOnlyAttribute.cs(9,43): error CS0518: Predefined type 'System.String' is not defined or imported [/home/dmg/projects/public/BenchmarkDotNet/
tests/BenchmarkDotNet.Tests/BenchmarkDotNet.Tests.csproj]
XUnit/TheoryWindowsOnlyAttribute.cs(9,9): error CS0518: Predefined type 'System.Void' is not defined or imported [/home/dmg/projects/public/BenchmarkDotNet/tes
ts/BenchmarkDotNet.Tests/BenchmarkDotNet.Tests.csproj]
    155 Warning(s)
    11220 Error(s)

Time Elapsed 00:00:13.92
```

Full build log:
https://gist.github.com/damageboy/09c715f9159f36d92813f31323d4b981

### PR DESCRIPTION
**Pull request by damageboy**
closes #1326

### CODE REVIEW

**Comment by adamsitnik** [comment]
@AndreyAkinshin  good point

We can also make it conditional to be included only for full framework projects

```xml
  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) ">
    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
    </PackageReference>
  </ItemGroup>
```

**Comment by damageboy** [comment]
@AndreyAkinshin Of course!
I was looking for a `Directory.Build.props` somewhere there, never occured to me you have something else going on there...

**Comment by damageboy** [comment]
@adamsitnik  @AndreyAkinshin 
I included the assemblies through common.props.
The two disassembler project were not including common.props so I added an `<Import>` to them.
The F# samples project doesn't like including common.props so I patched those refs manually in that single project file.

File: build/common.props [hunk 1]
```diff
@@ -57,9 +57,10 @@
...
-  </PropertyGroup>
-</Project>
\ No newline at end of file
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) ">
+    <PackageReference Include="Microsoft.NETFramework.ReferenceAssemblies" Version="1.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+</Project>
```
Comment (adamsitnik) [inline build/common.props]: @mfilippov could you please take a look? I want to make sure that we don't break your fix from #898

Comment (damageboy) [reply build/common.props]: Just to be clear. The previous way was the old way I was using in my own projects as well before the reference assemblies nuget became a reality.

In other words, I didn't just remove the old stuff unintentionally.

Comment (adamsitnik) [reply build/common.props]: > I didn't just remove the old stuff unintentionally.

Sure! I just want to make sure that the "new way" works fine for what @mfilippov needed in #898

File: build/common.props [hunk 2]
```diff
@@ -57,9 +57,10 @@
...
     <PackageVersion>$(Major).$(Minor).$(Revision).$(BuildNumber)$(PrereleaseLabel)</PackageVersion>
   </PropertyGroup>
-  <PropertyGroup>
-    <TargetFrameworkRootPath Condition="'$(TargetFrameworkRootPath)' == '' AND Exists('/Library/Frameworks/Mono.framework/Versions/Current')">/Library/Frameworks/Mono.framework/Versions/Current/lib/mono/xbuild-frameworks</TargetFrameworkRootPath>
-    <TargetFrameworkRootPath Condition="'$(TargetFrameworkRootPath)' == '' AND Exists('/usr/local/lib/mono/xbuild-frameworks')">/usr/local/lib/mono/xbuild-frameworks</TargetFrameworkRootPath>
-    <TargetFrameworkRootPath Condition="'$(TargetFrameworkRootPath)' == '' AND Exists('/usr/lib/mono/xbuild-frameworks')">/usr/lib/mono/xbuild-frameworks</TargetFrameworkRootPath>
-  </PropertyGroup>
-</Project>
\ No newline at end of file
+  <ItemGroup Condition=" $(TargetFramework.StartsWith('net4')) ">
```
Comment (mfilippov) [inline build/common.props]: You don't need condition here. It handled into ref asm packages.

Comment (damageboy) [reply build/common.props]: Force pushed and fixed!

File: build/common.props (hunk 1 above)
Comment (damageboy) [reply build/common.props]: @mfilippov I think @adamsitnik is waiting for you to confirm that this still works fine w.r.t to 898.

**Pull request merged**